refract-spec.md

```diff
@@ -117,29 +485,570 @@ This example will show a JSON object represented as both a normal Refract Elemen
+
+```json
+{
+  "element": "string",
+  "meta": {
+    "id": "foo"
+  },
   "content": "bar"
 }
 ```
 
-##### Compact Element Representation
+### Ref Element (Element)
+
+The `ref` element MAY be used to reference elements in remote documents or elements in the local document. The `ref` element transcludes the contents of the element into the document in which it is referenced.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (Namespace, required) - A URL or fragment identifier. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+
+
+#### Examples
+
+Elements MAY be referenced in remote or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "foo"
+}
+```
+
+##### Referencing Elements in Prefixed Namespace
+
+This references the element with the ID of `foo` in the prefixed namespace of `ns`.
+
+```json
+{
+  "element": "ref",
+  "content": {
+    "prefix": "ns",
+    "href": "foo"
+  }
+}
+```
+
+### Class Element (Element)
+
+The Class Element provides an element structure for giving classifications to things. The content of the class element MAY be either a string or an Element.
+
+#### Properties
+
+- `element` class (string, fixed)
+- `content` (array)
+    - (enum)
+        - (string)
+        - (Element)
+
+#### Examples
+
+This example shows a class element in use, where the classes given are "user" and "customer."
+
+```json
+{
+  "element": "class",
+  "content": ["user", "customer"]
+}
+```
+
+The Class Element MAY also be used as a meta attribute to give classifications to elements.
+
+```json
+{
+  "element": "string",
+  "meta": {
+    "class": ["name"]
+  },
+  "content": "John Doe"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. The following rules and constraints apply to namespaces.
+
+1. If there is a conflict, the namespaced element defined or referenced last MUST take precedence
+1. Types defined within the local namespace MUST take precedence over referenced namespaces
+1. Prefixed namespaces are only accessible through using a prefix
+1. Namespaces apply to the element in which they are introduced and all child elements
+1. Namespaces MUST not include the elements nor the data in the elements referenced, but rather include the constraints and conditions around the types defined within that namespace
+
+### Namespaces Element (Element)
+
+The `namespaces` element is a list of Namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
```

Comment:
As I said, I think could be something different than this that can be the definition of a ref content. In the ref context,
if it is a string, it is the current namespace and href is a relative fragment (with or without # depending on how we define it) points to local element. If href is absolute and no prefix, still has to point to local element. If it has prefix and fragment it lives that namespace, but if it is absolute URI it is that element etc. Transclude it. If it is in namespaces it references constraints.